android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -1321,9 +1322,29 @@ protected void showHelpBubble(Context context, float fx, float fy) {
       @Override
       public void onDismiss() {
         helpBubbleWindow = null;
+        helpBubblePos = null;
       }
     });
 
+    helpBubblePos = new float[]{px, py};
+
+    helpBubbleWindow.setAnimationStyle(R.style.PopupAnim);
+    if (getWindowToken() != null) {
+      this.repositionHelpBubble();
+    } else {
+      helpBubbleWindow = null;
```

Comment:
If `getWindowToken() == null` and we are throwing away that help bubble, surely we should never create `helpBubbleWindow` at all -- see above. In fact, AFAICT, by leaving it like this we'd leave a dangling window instance in this case, because `helpBubbleWindow` is never `.dismiss()`ed, just its reference is `null`ed here?

In other words, don't do anything with helpBubbleWindow at all in this function if `getWindowToken() == null`.

Reply:
In case it's not clear, this code was 100% unaltered from the original version.  Git just didn't recognize the change boundaries in the same way we would.  Please check and compare with the original lines from 1340-1346.

That said, there is a bit of distinction here - in the new code, there will be no dangling window instance.  That's because the new `reposition` is _also_ used to do _initial_ positioning - and that's what the call in the "then" section accomplishes.  Initial-location display code would perfectly match repositioned-location display code, so rather than go WET, I kept things DRY with that single method.

Reply:
... and it looks like I forgot to DRY out some of the logic by the final commit.  I should fix that real quick...

Reply:
Added a new commit to address this; that look fine?  (It's the only thing addressed by https://github.com/keymanapp/keyman/pull/6871/commits/4bf397009a13eb760fd0b759138722b2b7f297c7.)